File: 2020/csharp/Aoc2020.Tests/Day14Tests.cs
using System.Linq;
using System.Collections.Generic;
using Xunit;

namespace Aoc2020.Tests
{
    public class Day14Tests
    {
        [Fact]
        public void RunWorks()
        {
            Assert.Equal(165ul, new Computer().Run(Utils.ReadInputLines("14-test")).MemSum);
        }

        [Fact]
        public void Run2Works()
        {
            Assert.Equal(208ul, new Computer().Run2(Utils.ReadInputLines("14-test2")).MemSum);
        }

        [Fact]
        public void Stars()
        {
            var run = new Day14();
            Assert.Equal("2346881602152", run.Result1());
            Assert.Equal("3885232834169", run.Result2());
        }
    }
}

File: 2022/fsharp/Aoc2022.Tests/Day14.Tests.fs
module Day14.Tests

open Xunit
open Utils
open Day14

let input =
    """
    498,4 -> 498,6 -> 496,6
    503,4 -> 502,4 -> 502,9 -> 494,9
    """
    |> String.toLines

[<Fact>]
let ``Day14 simulate works`` () =
    Assert.Equal(24, input |> parse Part1 |> simulate)
    Assert.Equal(93, input |> parse Part2 |> simulate)

[<Fact>]
let ``Day14 Stars`` () =
    try
        Assert.Equal("757", job1 ())
        Assert.Equal("24943", job2 ())
    with :? System.NotImplementedException ->
        ()
